### PR TITLE
Updates to .travis.yml, for running python code

### PR DESCRIPTION
**Pull request by umesh-timalsina**
This PR changes `.travis.yml` to run the python code generated by `deepforge-keras`, creating a python3.5 runtime during install. To verify requirements, a test `requirements/python-requirements.spec.js` has been added.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+language: node_js
+services: mongodb
+sudo: true
+before_install:
+  - sudo apt-get update -q
+  - sudo apt-get install python3-pip python-dev -y
+  - pip3 install -U setuptools
+  - pip3 install -U virtualenvwrapper
+install:
+  - nvm install-latest-npm
```
Comment (brollb) [inline .travis.yml]: Why is this added? Are the versions for `node_js` ignored now?

File: test/requirements/python-requirements.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*jshint node:true, mocha:true*/
+
+'use strict';
+const { spawnSync } = require('child_process');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+describe('python-test', function () {
+    it('Should find python 3', (done) => {
```
Comment (brollb) [inline test/requirements/python-requirements.spec.js]: If the test is synchronous, `done` is not needed.

File: test/requirements/python-requirements.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        const processOutput = spawnSync(`python3`, ['-c', `"print('hello world')";`]);
+        assert(processOutput.status === 0);
```
Comment (brollb) [inline test/requirements/python-requirements.spec.js]: This is fine but `assert.equal(processOutput.status, 0)` might give a better default error message.

Comment (brollb) [reply test/requirements/python-requirements.spec.js]: (I still may have some asserts like this in the code base but have been trying to move over to this style myself :) )

File: test/requirements/python-requirements.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        done();
+    });
+
+    describe('requirements-test', function() {
+        let requirements = fs.readFileSync(path.join(__dirname, '..', '..', 'requirements.txt'),
+            {encoding: 'utf-8'}).split('\n').filter((val) => val);
+        requirements.forEach((req) => {
+            it(`Should be able to import package ${req}`, (done) =>{
+                const processOutput = spawnSync(`python3`, ['-c', `"import ${req}";`]);
+                assert(processOutput.status === 0);
```
Comment (brollb) [inline test/requirements/python-requirements.spec.js]: Same comments as above for `done` and `assert`

File: test/requirements/python-requirements.spec.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*jshint node:true, mocha:true*/
+
+'use strict';
+const { spawnSync } = require('child_process');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+describe('python-test', function () {
```
Comment (brollb) [inline test/requirements/python-requirements.spec.js]: No need to add the `-test` suffix (that part is obvious ;) )

File: test/requirements/python-requirements.spec.js (hunk 2 above)
Comment (brollb) [reply test/requirements/python-requirements.spec.js]: Also, "should" is generally lowercase when using mocha. Not a huge deal but might look a little funny.

File: test/requirements/python-requirements.spec.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    it('Should find python 3', (done) => {
+        const processOutput = spawnSync(`python3`, ['-c', `"print('hello world')";`]);
+        assert(processOutput.status === 0);
+        done();
+    });
+
+    describe('requirements-test', function() {
+        let requirements = fs.readFileSync(path.join(__dirname, '..', '..', 'requirements.txt'),
+            {encoding: 'utf-8'}).split('\n').filter((val) => val);
```
Comment (brollb) [inline test/requirements/python-requirements.spec.js]: encoding can be passed as the second argument to `readFileSync` and `.trim()` might be more concise if you are only worried about leading/trailing new lines. (If you are worried about intermediate empty lines, `.map(line => line.trim()).filter(line => !!line)` might be better.)

File: .travis.yml [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
+  - npm install
+  - pip3 install -r requirements.txt
+node_js:
+  - "8"
+  - "10"
+python:
+  - "3.5"
```
Comment (brollb) [inline .travis.yml]: Why are you installing python manually and adding it here? Which one is actually used?

File: .travis.yml (hunk 1 above)
Comment (umesh-timalsina) [reply .travis.yml]: This is because the `npm` used to install packages was not working for `webgme-engine` installation. The error I got was (no module named `text`)

File: test/requirements/python-requirements.spec.js (hunk 2 above)
Comment (umesh-timalsina) [reply test/requirements/python-requirements.spec.js]: Will get rid of them.

File: test/requirements/python-requirements.spec.js (hunk 3 above)
Comment (umesh-timalsina) [reply test/requirements/python-requirements.spec.js]: Ok. That should be a standard practice of using assert.

File: test/requirements/python-requirements.spec.js (hunk 5 above)
Comment (umesh-timalsina) [reply test/requirements/python-requirements.spec.js]: Ok.

File: .travis.yml (hunk 1 above)
Comment (brollb) [reply .travis.yml]: gotcha. Why not `npm install -g npm`?

Comment (umesh-timalsina) [reply .travis.yml]: Our build with different `node` versions are managed using `nvm` by travis. I think we can certainly replace this with `npm install -g npm`.

File: .travis.yml (hunk 7 above)
Comment (umesh-timalsina) [reply .travis.yml]: Actually, python 3.5 is being used. Should get rid of `python-dev`, as might not be needed for any of our requirements.

File: .travis.yml (hunk 1 above)
Comment (umesh-timalsina) [reply .travis.yml]: https://travis-ci.org/deepforge-dev/deepforge-keras/builds/608312198. 
Travis build is passing with `npm` update using `npm`.

File: .travis.yml (hunk 7 above)
Comment (umesh-timalsina) [reply .travis.yml]: To verify that python3.5 is being selected, I have added a test like this:
```javascript
it('should find python 3.5', () => {
        const processOutput = spawnSync('python3', ['-c', `import sys; assert sys.version[0:3] == '3.5'`]);
        assert.equal(processOutput.status, 0);
    });
```
